crates/ty_python_semantic/src/types/signatures.rs

```diff
@@ -626,10 +658,13 @@ impl<'db> Signature<'db> {
                 EitherOrBoth::Right(_) => {
                     // If there are more parameters in `other` than in `self`, then `self` is not a
                     // subtype of `other`.
-                    return false;
+                    return Ok(false);
                 }
 
                 EitherOrBoth::Both(self_parameter, other_parameter) => {
+                    let self_type = self_parameter.annotated_type();
+                    let other_type = other_parameter.annotated_type();
```

Comment:
I think I'd prefer to not use this and instead just use `.annotated_type` method directly. It has the benefit that it's obvious what the type corresponds to (annotated parameter type) as compared to `self_type` which sounds a bit generic.